src/main/java/org/shanerx/tradeshop/util/JsonConfiguration.java

```diff
@@ -56,55 +60,40 @@ public JsonConfiguration(Chunk c) {
 				throw new RuntimeException(exception);
 			}
 		}
-	}
 
-	private void writeJSON(String path, String value) {
-		JsonObject obj = new JsonObject();
-		obj.add(path, new JsonParser().parse(value));
 		try {
-			FileWriter fileWriter = new FileWriter(this.file);
-			fileWriter.write(obj.getAsString());
-			fileWriter.flush();
-			fileWriter.close();
-		} catch (Exception e) {
-			throw new RuntimeException(e);
-		}
-	}
-
-	private JsonObject readJSON(String path) {
-		try {
-			return new JsonParser().parse(new FileReader(file)).getAsJsonObject();
+			jsonObj = new JsonParser().parse(new FileReader(file)).getAsJsonObject();
 		} catch (FileNotFoundException e) {
 			throw new RuntimeException(e);
 		}
 	}
 
-	// CAREFUL: DOES NOT OUTPUT JSON! USE ONLY FOR FILE NAMES!
-	private String serializeChunk(Chunk chunk) {
-		String world = chunk.getWorld().getName();
-		int x = chunk.getX(), z = chunk.getZ();
+	private void setShop(Shop shop) {
+		Gson gson = new GsonBuilder().setPrettyPrinting().serializeNulls().create();
```

Comment:
Not sure why pretty printing is necessary - we use JSON for persistent data storage functions, not configurations/user Interfaces, which means that, if the file size increases, performance will be affected because of all the whitespace and other chars. (It is still micro-performance, though, hence it is a matter of taste.)

Reply:
@SparklingComet didn't think about that, I still like it to be a bit more readable but it isn't necessary so I'll take it out

Reply:
Meh
It doesn’t matter much
If you prefer pretty printing, I don’t have any objections.